src/chainstate/stacks/db/blocks.rs

```diff
@@ -1069,6 +982,19 @@ impl StacksChainState {
         }
     }
 
+    /// Load up a preprocessed block from the staging DB, regardless of its processed status.
+    /// Do not load the associated block.
+    pub fn load_staging_block_info(
+        block_conn: &DBConn,
+        index_block_hash: &StacksBlockId,
+    ) -> Result<Option<StagingBlock>, Error> {
+        let sql = "SELECT * FROM staging_blocks WHERE index_block_hash = ?1 AND orphaned = 0";
+        let args: &[&dyn ToSql] = &[&index_block_hash];
+        let mut rows =
+            query_rows::<StagingBlock, _>(block_conn, sql, args).map_err(Error::DBError)?;
```

Comment:
This should use one of the `query_row` functions -- it avoids allocating an uneccessary `Vec`

Reply:
can do!

Reply:
Addressed in ee12847